### PR TITLE
Disable git safe repo check and optimize behavior

### PR DESCRIPTION
**Pull request by rcdailey**
List of changes:

- Add global config option to make our clone directory "safe". Without
  this, the github action fails completely due to some security features
  implemented in the newest version of git.
- Consolidated a lot of git operations toward the bottom of the
  entrypoint script to avoid popping in & out of the clone directory
  more than needed.
- Use `clone` instead of `init` + `pull` for simpler logic and automatic
  `origin` remote to make later operations simpler.
- The final `git push` command has been simplified greatly.

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -28,17 +28,12 @@ if [ -z "$WIKI_DIR" ]; then
...
-git config user.name $GH_NAME
-git config user.email $GH_MAIL
+# Disable Safe Repository checks
+git config --global --add safe.directory "/github/workspace"
+git config --global --add safe.directory "/github/workspace/$TEMP_CLONE_FOLDER"
 
-git pull https://$GH_TOKEN@github.com/$REPO.wiki.git
-cd ..
+echo "Cloning wiki git..."
+git clone https://$GH_TOKEN@github.com/$REPO.wiki.git $TEMP_CLONE_FOLDER
```
Comment (Andrew-Chen-Wang) [inline entrypoint.sh:36]: I believe this should include both the GH_TOKEN and GH_NAME:GH_TOKEN as seen in the bottom of the file. We've had issues in the past when it was just the GH_TOKEN.

File: entrypoint.sh [hunk 2]
```diff
@@ -28,17 +28,12 @@ if [ -z "$WIKI_DIR" ]; then
...
-echo "Configuring wiki git..."
-mkdir $TEMP_CLONE_FOLDER
-cd $TEMP_CLONE_FOLDER
-git init
-
-# Setup credentials
-git config user.name $GH_NAME
-git config user.email $GH_MAIL
+# Disable Safe Repository checks
+git config --global --add safe.directory "/github/workspace"
```
Comment (Andrew-Chen-Wang) [inline entrypoint.sh:32]: is this the same as the "Verified" sign attached to git commits? For instance, you'll see verified and unsafe commit checks in a commit history

**Review by Andrew-Chen-Wang — CHANGES_REQUESTED** [review]

**Comment by Andrew-Chen-Wang** [comment]
Thanks a ton for this vital update!

File: entrypoint.sh (hunk 1 above)
Comment (rcdailey) [reply entrypoint.sh:36]: I'm happy to add that, but I kept the URL exactly as you already had it for the `git pull` command. The same URL is also used for the `push` performed later on. Your previous code attempted to use both for the push, but I feel that only one should be used to keep things simple. I'll update it to include both as you ask.

File: entrypoint.sh (hunk 2 above)
Comment (rcdailey) [reply entrypoint.sh:32]: No it's not the same. As of (I believe) Git v2.35, a security vulnerability was addressed that resulted in a new [safe.directory config](https://git-scm.com/docs/git-config#Documentation/git-config.txt-safedirectory). Basically unless you put your repo path in this config setting, git operations you run will fail in this repo if the ownership of that directory does not match the current user. So it's just a filesystem permission check, has nothing to do with commits AFAIK.

This was a very recent change, as I understand it, so you probably wouldn't have seen any issue until recently (I don't know when Github upgrades the version of git they use internally for build agents).

File: entrypoint.sh (hunk 1 above)
Comment (rcdailey) [reply entrypoint.sh]: Ok change is made. Here is the Github Action I use in another personal project of mine to test my changes:
https://github.com/rcdailey/trash-updater/runs/6162111655?check_suite_focus=true

**Review by Andrew-Chen-Wang — CHANGES_REQUESTED** [review]

File: entrypoint.sh [hunk 3]
```diff
@@ -28,17 +28,12 @@ if [ -z "$WIKI_DIR" ]; then
...
+git config --global --add safe.directory "/github/workspace/$TEMP_CLONE_FOLDER"
 
-git pull https://$GH_TOKEN@github.com/$REPO.wiki.git
-cd ..
+echo "Cloning wiki git..."
+git clone https://$GH_NAME:$GH_TOKEN@github.com/$REPO.wiki.git $TEMP_CLONE_FOLDER
```
[the remark this answers is not in the document]
Comment (rcdailey) [reply entrypoint.sh:36]: Why are two attempts needed? This seems unnecessary to me

[the remark this answers is not in the document]
Comment (Andrew-Chen-Wang) [reply entrypoint.sh:36]: There was an issue before that required this addition: https://github.com/Andrew-Chen-Wang/github-wiki-action/commit/b386aca0ddc5ec22b6003ba4cb50fa0b17243f6c

[the remark this answers is not in the document]
Comment (Andrew-Chen-Wang) [reply entrypoint.sh:36]: oh wait i'm stupid. THis is the cloning

**Comment by rcdailey** [comment]
Thank you for your time and quick turn around!

**Pull request merged**